### PR TITLE
Deploy to Lisk

### PR DESCRIPTION
**Pull request by ezynda3**
# Which Jira task belongs to this PR?

# Why did I implement it this way?

# Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] This pull request is as small as possible and only tackles one problem
- [ ] I have added tests that cover the functionality / test the bug
- [ ] I have updated any required documentation

# Checklist for reviewer (DO NOT DEPLOY and contracts BEFORE CHECKING THIS!!!)

- [ ] I have checked that any arbitrary calls to external contracts are validated and or restricted
- [ ] I have checked that any privileged calls (i.e. storage modifications) are validated and or restricted
- [ ] I have ensured that any new contracts have had AT A MINIMUM 1 preliminary audit conducted on <date> by <company/auditor>

### CODE REVIEW

File: config/networks.json [hunk 1]
```diff
@@ -323,6 +323,24 @@
...
+    "type": "mainnet",
+    "rpcUrl": "https://rpc.api.lisk.com",
+    "verificationType": "blockscout",
+    "explorerUrl": "https://blockscout.lisk.com",
+    "explorerApiUrl": "https://blockscout.lisk.com/api",
+    "multicallAddress": "0xcA11bde05977b3631167028862bE2a173976CA11",
+    "safeApiUrl": "https://transaction-lisk.safe.optimism.io/api",
+    "safeAddress": "0x86E02d7383D6a045848b0a1A842996AC9E943113",
+    "safeWebUrl": "https://safe.optimism.io/home?safe=lisk:0x86E02d7383D6a045848b0a1A842996AC9E943113",
+    "gasZipChainId": 238
```
Comment (0xDEnYO) [inline config/networks.json:342]: If supported by GasZip then we should also deploy GasZip (both contracts) and whitelist as periphery

File: deployments/lisk.diamond.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
+{
+    "LiFiDiamond": {
+        "Facets": {
+            "0x57FdfF2e36De6c8a8Cde297B150Ae291132Eae8d": {
+                "Name": "DiamondCutFacet",
+                "Version": "1.0.0"
+            },
+            "0xA1a4d577709dC4A70CA38F1D41562fab3aD09D3f": {
+                "Name": "DiamondLoupeFacet",
+                "Version": "1.0.0"
```
Comment (0xDEnYO) [inline deployments/lisk.diamond.json:10]: I actually wonder how you can have the versions in the diamond logs if the deploy log does not contain that information....that is utterly weird...

File: config/amarok.json [hunk 3]
```diff
@@ -108,6 +108,12 @@
       "0xe5d7c2a44ffddf6b295a15c148167daaaf5cf34f"
     ]
   },
+  "lisk": {
+    "chainId": 1135,
+    "connextHandler": "0x0000000000000000000000000000000000000000",
+    "domain": "",
+    "tokensToApprove": []
+  },
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply config/amarok.json:116]: Not needed

File: config/stargate.json [hunk 4]
```diff
@@ -31,6 +31,7 @@
     "kaia": "0x0000000000000000000000000000000000000000",
     "velas": "0x0000000000000000000000000000000000000000",
     "linea": "0x1a44076050125825900e736c501f859c50fE728c",
+    "lisk": "0x0000000000000000000000000000000000000000",
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply config/stargate.json:34]: Not needed

File: deployments/lisk.diamond.json (hunk 2 above)
Comment (ezynda3) [reply deployments/lisk.diamond.json:10]: It's generated by the deploy tool

File: script/deploy/_targetState.json [hunk 5]
```diff
@@ -1317,49 +1281,20 @@
...
-        "LiFuelFeeCollector": "1.0.1",
-        "TokenWrapper": "1.0.0",
-        "LiFiDEXAggregator": "1.0.0",
-        "Permit2Proxy": "1.0.0",
-        "SymbiosisFacet": "1.0.0"
+        "LiFiDEXAggregator": "1.0.0"
       }
     }
   },
+  "rootstocks": {},
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/deploy/_targetState.json]: fixed

File: config/networks.json [hunk 6]
```diff
@@ -323,6 +323,24 @@
     "safeWebUrl": "https://app.safe.global/transactions/queue?safe=linea:0xdf61270fDC1A892874Fd3C0143A0A4CBA74F4EF1",
     "gasZipChainId": 59
   },
+  "lisk": {
+    "name": "Lisk",
+    "chainId": 1135,
+    "nativeAddress": "0x0000000000000000000000000000000000000000",
+    "nativeCurrency": "ETH",
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply config/networks.json:330]: it's op stack now so this is not true

File: script/deploy/_targetState.json [hunk 7]
```diff
@@ -959,6 +958,35 @@
       }
     }
   },
+  "fraxtal": {
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/deploy/_targetState.json]: fixed

File: deployments/_deployments_log_file.json [hunk 8]
```diff
@@ -690,6 +690,34 @@
...
+          }
+        ]
+      }
+    },
+    "lisk": {
+      "production": {
+        "2.0.0": [
+          {
+            "ADDRESS": "0x57FdfF2e36De6c8a8Cde297B150Ae291132Eae8d",
+            "OPTIMIZER_RUNS": 0,
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply deployments/_deployments_log_file.json]: fixed

File: deployments/_deployments_log_file.json [hunk 9]
```diff
@@ -22492,6 +22996,20 @@
...
+    "lisk": {
+      "production": {
+        "1.0.1": [
+          {
+            "ADDRESS": "0xBa713B18c806EcdEEE49FAec623dE2D872192872",
+            "OPTIMIZER_RUNS": "1000000",
+            "TIMESTAMP": "2024-11-25 10:30:44",
+            "CONSTRUCTOR_ARGS": "0x0000000000000000000000000000000000000000000000000000000000000000",
+            "SALT": "",
+            "VERIFIED": "false"
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply deployments/_deployments_log_file.json]: fixed

File: script/utils/fetch-missing-deployments.ts [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+    for (const [contractName, contractAddress] of Object.entries(deployments)) {
+      try {
+        // Call explorer API
+        const url = new URL(networkConfig.explorerApiUrl)
+        url.searchParams.append('module', 'contract')
+        url.searchParams.append('action', 'getsourcecode')
+        url.searchParams.append('address', contractAddress as string)
+        url.searchParams.append('apiKey', apiKey)
+
+        const response = await fetch(url.toString())
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/utils/fetch-missing-deployments.ts:46]: not an issue with newer versions of node

File: script/utils/fetch-missing-deployments.ts [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+              VERIFIED: true,
+            },
+          ]
+          console.log('Updated')
+        } else {
+          console.log('Entry already exists')
+        }
+      } catch (error) {
+        console.error(`Error processing ${contractName}:`, error)
+        continue
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/utils/fetch-missing-deployments.ts]: fixed

File: deployments/_deployments_log_file.json [hunk 12]
```diff
@@ -22492,6 +22492,20 @@
           }
         ]
       }
+    },
+    "lisk": {
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply deployments/_deployments_log_file.json:23000]: Fixed

File: script/utils/fetch-missing-deployments.ts [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    // Write updated master log
+    fs.writeFileSync(masterLogPath, JSON.stringify(masterLog, null, 2))
+    console.log(`Successfully updated deployment logs for ${network}`)
+  } catch (error) {
+    console.error('Error:', error)
+    process.exit(1)
+  }
+}
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/utils/fetch-missing-deployments.ts]: not needed

**Review by 0xDEnYO — APPROVED** [review]

**Review by 0xDEnYO — APPROVED** [review]

**Pull request merged**